iModelCore/ECDb/ECDb/ECSql/PropertyNameExp.cpp

```diff
@@ -381,6 +388,15 @@ BentleyStatus PropertyNameExp::ResolveColumnRef(ECSqlParseContext& ctx)
      }
 
     if (matchProps.empty()) {
+        // if it's RelECClassId argument of NAV function then assume that it's a static variable
+        if (GetParent()->GetType() == Exp::Type::NavValueCreationFunc && GetParent()->GetAs<NavValueCreationFuncExp>().GetRelECClassIdExp() == this)

```

Comment:
I do not think we are making sure property path has as many elements as we require here.

Reply:
We don't need this piece of code, removed